### ISSUE TITLE
Cassette file keeps doubling with record set to none

### ISSUE THREAD

**Issue by SirNerdBear** [vcr/vcr#829]
I've searched everywhere and can't find any information on this. Maybe I am missing something. But each time my test suite runs it is appending the cassette even though I've set record to none. 

The result is my cassette file grows to massive megabyte sizes very quickly. 

```
VCR.configure do |config|
  config.cassette_library_dir = "test/tapes"
  config.hook_into :webmock
  config.define_cassette_placeholder("<TOKEN>") { Rails.application.credentials.cloudflare[:token] }
  
  #config.debug_logger = $stderr
  config.default_cassette_options = {
    allow_unused_http_interactions: true,
    record: :none,
    :allow_playback_repeats => true,
    record_on_error: false,
    match_requests_on: [:uri, :body, :method] 
  }
end
```

**Comment by nicolasiensen** [comment]
Could you share the implementation of the test that creating and amending the cassette file?

**Comment by SirNerdBear** [comment]
Sure

```
  it "Updates subdomain" do
        VCR.use_cassette("api.cloudflare.com", :record => :none) do
            @record.save
        end
  end
```

The model being saved has an :after_save hook which calls the Cloudflare API.

It definitely records the API interaction and it plays them back successfully. However, the cassette file keeps being amended. 

I'm running on Rails 6.  Ruby 2.7.1.

**Comment by nicolasiensen** [comment]
Thanks for sharing this, could you also share the cassette it is creating, please?

**Comment by SirNerdBear** [comment]
[out.zip](https://github.com/vcr/vcr/files/5152270/out.zip)

A zip of one of the bigger ones. I've been deleting it from time to time and starting over when it becomes too big. 

It's about 1,500 lines of YML on the first run. Every run after that adds ~1,500 more lines. 

Thanks for the help!

**Comment by nicolasiensen** [comment]
I can't see anything wrong with your test or how you are setting up VCR, I'll try to reproduce it to narrow down where the problem might be.

If you can, you could try to isolate the bug on your side by creating a leaner setup without Rails and other dependencies but just VCR and the minimal necessary to make the CloudFront request you are triggering.

**Comment by SirNerdBear** [comment]
Ya will do. I'll try doing a different API call (a really basic one) as well as see if maybe it's a conflict with the Cloudflare API gem I am using.

**Comment by andrewle** [comment]
@SirNerdBear I'm really curious to find out if you figured out a way to resolve this problem?

**Comment by SirNerdBear** [comment]
Unfortionitly I did not. Because of a deadline for the project I ended up using a different approach and dropped VCR. I'm hoping next refactor I can find the time to dig into it.

I suspect it has something to do with the way the Cloudflare API gem was written to always use ASYNC.

**Comment by trevorturk** [comment]
I'm encountering the same (or a similar) issue and created an example minitest suite to demonstrate:

https://github.com/trevorturk/async-http-vcr

There is some additional information in a related issue here: https://github.com/socketry/protocol-http1/issues/11 but tl;dr if you re-run this minitest suite, one of the two tests uses a cassette that gets a new `request` block appended to it with each test run. This may be an issue with the HTTP library I'm using `async-http` which is using Fibers and may be confusing VCR somehow.

**Comment by trevorturk** [comment]
Note these very helpful PRs aim to solve the underlying issue, which the authors found in the webmock interaction: https://github.com/bblimke/webmock/pull/930 && https://github.com/bblimke/webmock/pull/931

**Comment by olleolleolle** [comment]
@trevorturk Hi, and thanks for all the good context added to this Issue!

https://github.com/bblimke/webmock/blob/master/CHANGELOG.md#3113 3.11.3 was released a while ago and had the underlying fix in it.

Can this be closed, with that?

**Comment by trevorturk** [comment]
Yep, apologies, I didn't bookmark this issue and neglected to return to report the issue appears to be fixed! Thank you!

**Comment by olleolleolle** [comment]
@trevorturk Appreciate the hints in any case!

Closing.
